### DISCUSSION TITLE
Adding blocks of content to page sections

### DISCUSSION THREAD

**Discussion by futurewebsites** [area17/twill#1182] [General]
Is it possible to add sections lets say we have pages with `<section class="white"></section>` and another `<section class="grey"></section>` 

And we wanted to add everything from the block editor to a white section then another set to the grey section?

**Comment by pboivin** [comment]
Hey @futurewebsites, I can think of a few different ways to approach this.

Something I like to do is to have 2 modules: `Pages` and `PageSections`. A page is just a container for sections, and sections have all the flexibility of a standalone module. This works well for me in cases where sections are complex and is better to focus on one at a time for content management.

On the other hand, if sections are quite small, you can probably go very far with a `Pages` module and a `Section` block.

Also, if sections are fixed, maybe you could go with a dedicated block editor for each section, using the new multiple block editor feature: https://github.com/area17/twill/pull/918

**Reply by futurewebsites** [reply]
So a PageSection would have a parent of page, a select perhaps for the section bg colour, a position field so it could be re-ordered then block editor for the content for instance ? 

and to feature things like articles or in my case products I can use buckets?
 - on that note lets say I import brands and products but wanted the ability to feature products on a brand page in a set position and format would this be better achieved using a repeater?

**Reply by futurewebsites** [reply]
Better still I think would be to have a sectionable polymorphic relation which would allow me to create sections for any module!

**Reply by pboivin** [reply]
For related items like brands and products, the go to field is probably a Browser. Also, Browsers support ordering if the pivot has a `position` column. But in your case, if you want to add extra information on the relationship (like `format`), you can use a Repeater to wrap the Browser + custom fields.
